Projects/Contracts/Sources/View/ContractTable.swift

```diff
@@ -9,82 +9,72 @@ import hCoreUI
     var body: some View {
         ContractBundleLoadingIndicator()
-
+        
         hSection {
-            PresentableStoreLens(
-                ContractStore.self,
-                getter: { state in
-                    getContractsToShow(for: state, filter: filter.nonemptyFilter(state: state))
-                }
-            ) { contracts in
-                ForEach(contracts, id: \.id) { contract in
-                    ContractRow(id: contract.id)
-                        .fixedSize(horizontal: false, vertical: true)
-                        .padding(.top, 15)
-                        .transition(.slide)
-                }
+            ForEach(contracts, id: \.id) { contract in
+                ContractRow(id: contract.id)
+                    .fixedSize(horizontal: false, vertical: true)
+                    .padding(.top, 15)
+                    .transition(.slide)
             }
         }
         .presentableStoreLensAnimation(.spring())
         .sectionContainerStyle(.transparent)
-
-        PresentableStoreLens(
-            ContractStore.self,
-            getter: { state in
-                return self.filter.nonemptyFilter(state: state).displaysActiveContracts
-            }
-        ) { displaysActiveContracts in
-            if displaysActiveContracts {
-                CrossSellingStack()
-
-                PresentableStoreLens(
-                    ContractStore.self,
-                    getter: { state in
-                        getContractsToShow(for: state, filter: .terminated(ifEmpty: .none))
-                    }
-                ) { terminatedContracts in
-                    if !terminatedContracts.isEmpty {
-                        hSection(header: hText(L10n.InsurancesTab.moreTitle)) {
-                            hRow {
-                                hText(L10n.InsurancesTab.terminatedInsurancesLabel)
-                            }
-                            .withCustomAccessory({
-                                Spacer()
-                                hText(String(terminatedContracts.count), style: .body)
-                                    .foregroundColor(hLabelColor.secondary)
-                                    .padding(.trailing, 8)
-                                StandaloneChevronAccessory()
-                            })
-                            .onTap {
-                                store.send(.openTerminatedContracts)
-                            }
+    
+        if contracts.contains(where: { $0.currentAgreement?.status == .active }) {
+            CrossSellingStack()
+        }
+        
+        hSection {
+            PresentableStoreLens(
+                ContractStore.self,
+                getter: { state in
+                    return state.terminatedContracs
+                }
+            ) { terminatedContracts in
+                if !terminatedContracts.isEmpty {
+                    hSection(header: hText(L10n.InsurancesTab.moreTitle)) {
```

Comment:
Maybe move this view to a new property and avoid this indent gore? Not super important but felt it would be much more clearer to make sense of what is going on here

Reply:
Yeah good call!